### PR TITLE
Tnt-43818 odd client hints inclusion

### PR DESCRIPTION
**Pull request by shandilya3**
This PR adds support for client hints.

## How Has This Been Tested?
This will be tested thoroughly once we have backend system ready to support. This pr will be merged only after testing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by dcottingham — CHANGES_REQUESTED** [review]
Looks good, just have a few comments.  Also, can you please rename anything named "browserPlatform" or "BROWSER_PLATFORMS" to just platform/platforms.  I know this naming existed before your PR but I find the terminology confusing since the platform doesn't have anything to do with the browser.

File: src/main/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollator.java [hunk 1]
```diff
@@ -101,24 +101,39 @@
     Map<String, Object> user = new HashMap<>();
     String userAgent = extractUserAgent(deliveryRequest);
     ClientHints clientHints = extractClientHints(deliveryRequest);
-    user.put(USER_BROWSER_TYPE, parseBrowserType(userAgent));
+    String browserInfo = getBrowserInfo(userAgent, clientHints);
+    user.put(USER_BROWSER_TYPE, parseBrowserType(browserInfo));
```
Comment (dcottingham) [inline src/main/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollator.java:105]: Let's just save the result of `parseBrowserType(browserInfo)` to a variable so we can more easily use it in the `parseBrowserVersion` call.

File: src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java [hunk 2]
```diff
@@ -74,10 +76,110 @@ public void testChromeLinux() {
...
+    clientHints.setBrowserUAWithFullVersion(
+        "\" Not A;Brand\";v=\"99.0.0.0\", \"Chromium\";v=\"99.0.4844.83\", \"Google Chrome\";v=\"99.0.4844.83\"");
+    String userAgent =
+        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.109 Safari/537.36";
+
+    TargetDeliveryRequest request = requestWithClientHintsAndUA(clientHints, userAgent);
+    Map<String, Object> result = collator.collateParams(request, pageLoad);
+    assertEquals("chrome", result.get(UserParamsCollator.USER_BROWSER_TYPE));
+    assertEquals("mac", result.get(UserParamsCollator.USER_PLATFORM));
+    assertNotEquals("98", result.get(UserParamsCollator.USER_BROWSER_VERSION));
```
Comment (dcottingham) [inline src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java:168]: @shandilya3 - I'm confused why this is 98 if we are using the client hint and not user agent.

Comment (shandilya3) [reply src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java:168]: this is a negative test/assertNotEquals.  It should be 99 for assertEquals positive test.

Comment (dcottingham) [reply src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java:168]: Ah, my mistake for missing that.  In that case I think this test is effectively the same as `testClientHintsBrowserUAFullVersionInChrome`.  Maybe we can get rid of one of them.

**Review by dcottingham — APPROVED** [review]
Nice work!

File: src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java (hunk 2 above)
Comment (5amfung) [reply src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java:168]: If we expect the value to be 99, wouldn't it be more accurate to just `assertEqual("99",...)`.  I'm confused too.

Comment (shandilya3) [reply src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java:168]: **assertNotEquals**("98", result.get(UserParamsCollator.USER_BROWSER_VERSION));

Comment (shandilya3) [reply src/test/java/com/adobe/target/edge/client/ondevice/collator/UserParamsCollatorTest.java:168]: Let me remove this test, it's causing confusion.  😃

**Pull request merged**